src/validation.h

```diff
@@ -724,7 +724,9 @@ class CChainState
 
     /** Replay blocks that aren't fully applied to the database. */
     bool ReplayBlocks(const CChainParams& params);
-    bool RewindBlockIndex(const CChainParams& params) LOCKS_EXCLUDED(cs_main);
+
+    /** Check if chain state needs to be redownloaded due to lack of witness data */
+    [[nodiscard]] bool NeedsRedownload(const CChainParams& params) const EXCLUSIVE_LOCKS_REQUIRED(cs_main);
```

Comment:
```diff
-    /** Check if chain state needs to be redownloaded due to lack of witness data */
-    [[nodiscard]] bool NeedsRedownload(const CChainParams& params) const EXCLUSIVE_LOCKS_REQUIRED(cs_main);
+    /** Whether the chain state needs to be reindexed due to lack of witness data. */
+    [[nodiscard]] bool NeedsReindex(const CChainParams& params) const EXCLUSIVE_LOCKS_REQUIRED(cs_main);
     bool LoadGenesisBlock(const CChainParams& chainparams);
```

Reply:
I left it as `NeedsRedownload` as that represents more accurately, what's needed. Restarting with `-reindex` is just the suggestion we make to the user that accomplishes the redownload. To me, `NeedsReindex` suggests merely rebuilding an index from what we have would be enough.

Updated the comment though.